Cargo.toml

```diff
@@ -10,63 +10,86 @@ repository = "https://github.com/dusk-network/wallet-cli"
 license = "MPL-2.0"
 exclude = [".github/*", ".gitignore"]
 
-[[bin]]
-name = "rusk-wallet"
-path = "src/bin/main.rs"
+[lib]
+crate-type = ["cdylib", "rlib"]
 
 [dependencies]
-clap = { version = "3.1", features = ["derive", "env"] }
-thiserror = "1.0"
 anyhow = "1.0"
-tokio = { version = "1.15", features = ["full"] }
-serde = { version = "1.0", features = ["derive"] }
-url = { version = "2", features = ["serde"] }
 async-trait = "0.1"
-block-modes = "0.8"
-serde_json = "1.0"
-hex = "0.4"
-tiny-bip39 = "0.8"
-crossterm = "0.23"
-rand_core = "0.6"
-requestty = "0.4.1"
-futures = "0.3"
-base64 = "0.13"
-crypto = "0.3"
-whoami = "1.2"
-blake3 = "1.3"
-sha2 = "0.10.7"
-toml = "0.5"
-open = "2.1"
-dirs = "4.0"
-bs58 = "0.4"
+clap = { version = "4.4", features = ["derive", "env"] }
+crossterm = { version = "0.27", optional = true }
+dirs = { version = "5.0", optional = true }
 rand = "0.8"
-aes = "0.7"
-rocksdb = "0.21"
-flume = "0.10.14"
-reqwest = { version = "0.11", features = ["stream"] }
+requestty = { version = "0.5", optional = true }
+serde = "1.0"
+serde_json = "1.0"
+tokio = { version = "1.32", optional = true }
+url = { version = "2.4", optional = true }
+wasmer = "3.1"
+zeroize = { version = "1.6", features = ["derive"] }
+tracing = "0.1"
+tracing-subscriber = { version = "0.3", optional = true, features = ["fmt", "env-filter", "json"] }
+
+## temporary dependencies for mock wallet implementation
+bs58 = { version = "0.5", optional = true, default-features = false, features = ["alloc", "cb58"] }
+dusk-bytes = { version = "^0.1", optional = true }
+dusk-jubjub = { version = "0.12", optional = true }
+dusk-pki = { version = "0.12", optional = true, features = ["rkyv-impl"] }
+dusk-wallet-core = { git = "https://github.com/dusk-network/wallet-core", optional = true }
+phoenix-core = { version = "0.20.0-rc.0", optional = true, default-features = false, features = ["alloc", "rkyv-impl"] }
+poseidon-merkle = { version = "0.2.1-rc.0", optional = true, features = ["rkyv-impl"] }
+rusk-abi = { version = "0.10.0-piecrust.0.6", optional = true, default-features = false }
+rkyv = { version = "^0.7", optional = true, default-features = false, features = ["validation"] }
```

Comment:
These are added as `[dependencies]`. I was under the impression they were supposed to be eliminated entirely. I'm alright with them being under `[dev-dependencies]` but not here. Unless I'm wrong and this is not the intent, then I'm open to this.